cfme/automate/service_dialogs.py

```diff
@@ -77,166 +22,207 @@
+    title = Text('#explorer_title_text')
+
+    plus_btn = Dropdown('Add')
+    label = Input(name='label')
+    description = Input(name="description")
+
+    submit_button = Checkbox(name='chkbx_submit')
+    cancel_button = Checkbox(name='chkbx_cancel')
+
+
+class DialogsView(AutomateCustomizationView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and
+            self.title.text == 'All Dialogs' and
+            self.service_dialogs.is_opened and
+            self.service_dialogs.tree.currently_selected == ["All Dialogs"])
+
+
+class AddDialogView(DialogForm):
+
+    add_button = Button("Add")
+    plus_btn = Dropdown('Add')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == "Adding a new Dialog [Dialog Information]"
+        )
+
+
+class EditDialogView(DialogForm):
+    element_tree = ManageIQTree('dialog_edit_treebox')
+
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == "Editing Dialog {}".format(self.label)
+        )
 
-    def __init__(self, label=None, description=None,
-                 submit=False, cancel=False,
-                 tab_label=None, tab_desc=None,
-                 box_label=None, box_desc=None, element_data=None, appliance=None):
+
+class DetailsDialogView(AutomateCustomizationView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == 'Dialog "{}"'.format(self.context['object'].label)
+        )
+
+
+class DialogCollection(Navigatable):
+    """Collection object for the :py:class:`Dialog`."""
+
+    def __init__(self, appliance=None):
         Navigatable.__init__(self, appliance=appliance)
+
+    tree_path = ['All Dialogs']
+
+    def instantiate(self, label, description, submit=False, cancel=False):
+        return Dialog(
+            label=label, description=description, submit=submit, cancel=cancel,
+            appliance=self.appliance)
+
+    def create(self, label=None, description=None, submit=False, cancel=False):
+        """ Create dialog label method """
+        view = navigate_to(self, 'Add')
+        fill_dict = {
+            k: v
+            for k, v in {'label': label, 'description': description,
+            'submit_button': submit, 'cancel_button': cancel}.items()
+            if v is not None}
+        view.fill(fill_dict)
+        return Dialog(
+            label=label, description=description, submit=submit, cancel=cancel,
+            appliance=self.appliance)
+
+
+class Dialog(Navigatable, Fillable):
```

Comment:
Fillable objects should implement as_fill_value method that returns the name how the obejct appears in selections in the UI, I think in this case the label.